File: asn1runtime/src/main/scala/org/asn1gen/runtime/AsnReal.scala
package org.asn1gen.runtime

class AsnReal(val value: Double) extends AsnType {
  override def _desc: meta.AsnReal = meta.AsnReal
  
  def _raw: this.type = this

  def copy(value: Double = this.value): AsnReal = AsnReal(value)

  override def equals(that: Any): Boolean = that match {
    case that: AsnReal => this.value == that.value
    case _ => false
  }

  override def hashCode(): Int = this.value.hashCode

  def value(f: (Double => Double)): AsnReal = this.copy(value = f(this.value))
}

object AsnReal extends AsnReal(0.0) {
  def apply(value: Double) = new AsnReal(value)

  def unapply(value: AsnReal) = Some(value.value)
}

File: asn1jrt/src/main/java/org/asn1gen/runtime/java/AsnReal.java
package org.asn1gen.runtime.java;

public class AsnReal implements AsnType {
  public static AsnReal EMPTY = new AsnReal(0.0);
  
  public final double value;
  
  public AsnReal(final double value) {
    this.value = value;
  }
}
